### PR TITLE
VFB-158: Fetch statuses from database instead of from list

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Fetch statuses from status_order table instead of predetermined list from front end.

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| paste_here | paste_here |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database... N/A
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [x] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [x] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES / NO (remove as appropriate)
  - [x] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

File: src/databaseUtils.ts [hunk 1]
```diff
@@ -27,3 +27,5 @@ export type ViewSchema = {
 export type TableNames = keyof Tables;
 
 export type ViewNames = keyof Views;
+
+export type ParcelStatus = Schema["status_order"]["event_name"];
```
Comment (minalui) [inline src/databaseUtils.ts:31]: Not sure if this is the right place to put it

File: src/app/parcels/ActionBar/Statuses.tsx [hunk 2]
```diff
@@ -89,6 +71,25 @@ const Statuses: React.FC<Props> = ({
...
+
+    useEffect(() => {
+        const getParcelStatuses = async (): Promise<void> => {
+            const { data: parcelStatusesData, error: parcelStatusesError } =
+                await fetchParcelStatuses();
+            if (parcelStatusesError) {
+                switch (parcelStatusesError.type) {
+                    case "failedToFetchStatuses":
+                        setModalError(
+                            `Unable to retrieve statuses filter for parcels. Log ID: ${parcelStatusesError.logId}`
```
Comment (minalui) [inline src/app/parcels/ActionBar/Statuses.tsx:84]: I'm not sure where to display the error but modal error appears next to the status dropdown

File: src/databaseUtils.ts (hunk 1 above)
Comment (rickaw) [reply src/databaseUtils.ts:31]: This is good!

**Review by rickaw — COMMENTED** [review]
I think the task in this ticket was missed
> Change the events table’s event_name to be a FK to status_order called new_parcel_status

File: src/app/parcels/ActionBar/Statuses.tsx (hunk 2 above)
Comment (rickaw) [reply src/app/parcels/ActionBar/Statuses.tsx:84]: Yeah I think that's fine for now

File: supabase/seed/seed.mts [hunk 3]
```diff
@@ -87,23 +87,27 @@ async function generateSeed(): Promise<void> {
...
-                timestamp: (ctx) => getPseudoRandomDateBetween(earliestDate, latestDate, ctx.seed),
-            }),
-        { connect: true }
-    );
+    for (const status of eventNamesWithNoData) {
+        await seed.events(
+            (generate) =>
+                generate(1000, {
+                    newParcelStatus: status,
+                    eventData: (ctx) => copycat.int(ctx.seed, { min: 1, max: 10 }).toString(),
```
Comment (rickaw) [inline supabase/seed/seed.mts:107]: ( this is an event with no data)
```suggestion
                    eventData: (ctx) => "",
```

**Review by rickaw — APPROVED** [review]

File: supabase/seed/seed.mts (hunk 3 above)
Comment (minalui) [reply supabase/seed/seed.mts:107]: Good spot thank you 😥

**Review by rickaw — APPROVED** [review]

**Pull request merged**